lib/utils/aws/aws.go

```diff
@@ -484,3 +490,30 @@ func iamResourceARN(partition, accountID, resourceType, resourceName string) str
 		Resource:  fmt.Sprintf("%s/%s", resourceType, resourceName),
 	}.String()
 }
+
+// MaybeHashRoleSessionName truncates the role session name and adds a hash
+// when the original role session name is greater than AWS character limit
+// (64).
+func MaybeHashRoleSessionName(roleSessionName string) (ret string) {
+	if len(roleSessionName) <= MaxRoleSessionName {
+		return roleSessionName
+	}
+
+	hash := sha1.New()
+	hash.Write([]byte(roleSessionName))
+	hex := hex.EncodeToString(hash.Sum(nil))
```

Comment:
Suggestion: Make the hash smaller? Aesthetic only.

```suggestion
	const hashLen = 10
	hex := hex.EncodeToString(hash.Sum(nil))[:hashLen]
```